tests/components/Popover.spec.tsx

```diff
@@ -65,104 +71,129 @@ describe('<Popover>', () => {
+        });
+
+        it('should call the onToggle prop with false when calling toggleOpened with isOpen: false', () => {
+          (popoverWrapper.instance() as Popover).toggleOpened(false);
+
+          expect(onToggleSpy.calls.count()).toBe(1);
+
+          expect(onToggleSpy).toHaveBeenCalledWith(false);
+        });
+      });
     });
-  });
 
-  describe('Document click handler for an opened popover', () => {
-    beforeEach(() => {
-      popoverProps = _.extend(popoverProps, {
-        isOpen: true
+    describe('Tether toggle click handler', () => {
+      beforeEach(() => {
+        mountPopover();
       });
 
-      popoverWrapper = mount(
-        <Popover {...popoverProps} >
-          <span id='PopoverToggle'>Toggle</span>
-          <span id='PopoverElement'>Tether element</span>
-        </Popover>,
-        { attachTo: document.getElementById('App') }
-      );
-    });
+      it('should open the popover on click toggle if popover was closed', () => {
+        $(popoverSelector).click();
 
-    afterEach(() => {
-      popoverWrapper.unmount();
-      popoverWrapper.detach();
-    });
+        expect(toggleOpenedSpy.calls.count()).toBe(1);
+
+        expect(toggleOpenedSpy).toHaveBeenCalledWith(true);
+      });
 
-    it('should not close the popover on click tether element', () => {
-      expect(popoverWrapper.state('isOpen')).toBe(true);
+      it('should close the popover on click toggle if popover was opened', () => {
+        // First, open the Popover
+        $(popoverSelector).click();
 
-      $('#PopoverElement').click();
+        $(popoverSelector).click();
 
-      expect(popoverWrapper.state('isOpen')).toBe(true);
+        expect(toggleOpenedSpy.calls.count()).toBe(2);
+
+        expect(toggleOpenedSpy).toHaveBeenCalledWith(false);
+      });
     });
 
-    it('should close the popover when clicking outside Popover', () => {
-      expect(popoverWrapper.state('isOpen')).toBe(true);
+    describe('Document click handler for an opened popover', () => {
+      beforeEach(() => {
+        popoverProps = _.extend(popoverProps, {
+          isOpen: true
+        });
 
-      $('body').click();
+        mountPopover();
+      });
 
-      expect(popoverWrapper.state('isOpen')).toBe(false);
-    });
-  });
+      it('should not close the popover on click tether element', () => {
+        $('#PopoverElement').click();
 
-  describe('Document click handler for a closed popover', () => {
-    beforeEach(() => {
-      popoverProps = _.extend(popoverProps, {
-        isOpen: false
+        expect(toggleOpenedSpy.calls.count()).toBe(0);
       });
 
-      popoverWrapper = mount(
-        <Popover {...popoverProps}>
-          <span id='PopoverToggle'>Toggle</span>
-          <span id='PopoverElement'>Tether element</span>
-        </Popover>,
-        { attachTo: document.getElementById('App') }
-      );
-    });
+      it('should close the popover when clicking outside Popover', () => {
+        $('body').click();
```

Comment:
I am not sure, but you might be able to remove jQuery and use `popoverWrapper.closest('body').simulate('click')`, but it might not work while exiting the React component.